### PR TITLE
Filter command aliases for PHP keywords

### PR DESCRIPTION
**Pull request by damiankloip**
I was testing out something for one of the PsySH PRs and we have a problem with aliases. There is a high chance one of them could conflict with a PHP keyword. E.g. `if` is an alias of `image-flush` this means we can't eval any php code starting with if :)

Maybe we should check commands too to be safe?

Needs a bit of tidy, just trying to post ASAP before my laptop dies (on 2% atm).

### CODE REVIEW

File: lib/Drush/Psysh/DrushCommand.php [hunk 1]
```diff
@@ -118,7 +125,11 @@ protected function execute(InputInterface $input, OutputInterface $output) {
    *   The command aliases.
    */
   protected function buildAliasesFromConfig() {
-    return !empty($this->config['aliases']) ? $this->config['aliases'] : [];
+    $aliases = !empty($this->config['aliases']) ? $this->config['aliases'] : [];
+
+    return array_filter($aliases, function($alias){
```
Comment (weitzman) [inline lib/Drush/Psysh/DrushCommand.php]: Could array_diff() simplify this a little?

Comment (damiankloip) [inline lib/Drush/Psysh/DrushCommand.php]: Totally. When I plug my laptop in and have some power i'll carry on with tidying this up.

File: commands/core/cli.drush.inc [hunk 2]
```diff
@@ -70,14 +70,98 @@ function drush_cli_core_cli() {
...
   $ignored_commands = ['help', 'drush-psysh', 'php-eval', 'core-cli', 'php'];
+  $php_keywords = _drush_core_cli_get_php_keywords();
 
   foreach ($commands as $name => $config) {
-    // Ignore some commands that don't make sense inside PsySH, are hidden, or
-    // are aliases.
-    if (in_array($name, $ignored_commands) || !empty($config['hidden']) || ($name !== $config['command'])) {
+    // Ignore some commands that don't make sense inside PsySH, are PHP keywords
+    // are hidden, or are aliases.
+    if (in_array($name, $ignored_commands) || in_array($name, $php_keywords) || !empty($config['hidden']) || ($name !== $config['command'])) {
```
Comment (damiankloip) [inline commands/core/cli.drush.inc]: @weitzman this just removes the command if it's conflicts, we could potentially prefix it with drush or something? This is very edge-casey, not sure if it's worth doing that or not?

File: commands/core/cli.drush.inc [hunk 3]
```diff
@@ -70,14 +70,98 @@ function drush_cli_core_cli() {
...
       unset($commands[$name]);
     }
+    else {
+      // Make sure the command aliases don't contain any PHP keywords.
+      if (!empty($config['aliases'])) {
+        $commands[$name]['aliases'] = array_diff($commands[$name]['aliases'], $php_keywords);
```
Comment (damiankloip) [inline commands/core/cli.drush.inc]: filters out any aliases like `if`.

**Comment by damiankloip** [comment]
I prefer this approach, filtering the data before creating the commands, rather than having this logic in the actual command class.

**Comment by weitzman** [comment]
Looks good.

File: commands/core/cli.drush.inc (hunk 2 above)
Comment (dawehner) [inline commands/core/cli.drush.inc]: which commands are that?

File: commands/core/cli.drush.inc (hunk 3 above)
Comment (dawehner) [inline commands/core/cli.drush.inc]: seems reasonable, but should afaik be done ideally by pysh itself

File: commands/core/cli.drush.inc (hunk 2 above)
Comment (damiankloip) [inline commands/core/cli.drush.inc]: A good example is the alias for image-flush, if. Some of those keywords could easily clash with an alias or custom command.

File: commands/core/cli.drush.inc (hunk 3 above)
Comment (damiankloip) [inline commands/core/cli.drush.inc]: Yes, we could try to get this upstream. Let's keep it here for now though, as we can have this working before the 8.0.0 release, if we wait for upstream, we will likely not.

**Comment by dawehner** [comment]
for now +1

**Comment by damiankloip** [comment]
Thanks both, merging as-is so we have this fix in before 8.0.0.

**Pull request merged**